progenitor-impl/src/httpmock.rs

```diff
@@ -173,79 +173,136 @@ impl Generator {
                 };
 
                 let name_ident = format_ident!("{}", name);
-                let handler = match kind {
+                let (required, handler) = match kind {
                     OperationParameterKind::Path => {
                         let re_fmt = method.path.as_wildcard_param(api_name);
-                        quote! {
-                            let re = regex::Regex::new(
-                                &format!(#re_fmt, value.to_string())
-                            ).unwrap();
-                            Self(self.0.path_matches(re))
-                        }
+                        (
+                            true,
+                            quote! {
+                                let re = regex::Regex::new(
+                                    &format!(#re_fmt, value.to_string())
+                                ).unwrap();
+                                Self(self.0.path_matches(re))
+                            },
+                        )
                     }
-                    OperationParameterKind::Query(true) => quote! {
-                        Self(self.0.query_param(#name, value.to_string()))
-                    },
-
-                    OperationParameterKind::Query(false) => {
-                        // If the type is a ref, augment it with a lifetime that we'll also use in the function
-                        let (lifetime, arg_type_name) = if let syn::Type::Reference(mut rr) =
-                            syn::parse2::<syn::Type>(arg_type_name.clone()).unwrap()
-                        {
-                            rr.lifetime =
-                                Some(syn::Lifetime::new("'a", proc_macro2::Span::call_site()));
-                            (Some(quote! { 'a, }), rr.to_token_stream())
-                        } else {
-                            (None, arg_type_name)
-                        };
+                    OperationParameterKind::Query(true) => (
+                        true,
+                        quote! {
+                            Self(self.0.query_param(#api_name, value.to_string()))
```

Comment:
note this was previously using the sanitized name rather than the name from the API (i.e. it was using the wrong name)